lib/decorators/component.ts

```diff
@@ -51,9 +51,17 @@ import directiveControllerFactory from '../util/directive-controller';
 import {writeMapMulti} from './input-output';
 import {inputsMap} from '../properties/inputs-builder';
 import events from '../events/events';
+import {createConfigErrorMessage} from '../util/helpers';
 
 const TYPE = 'component';
 
+export const componentHooks = {
+	after: [],
+	extendDDO: [],
+	beforeCtrlInvoke: [],
+	afterCtrlInvoke: []
+};
+
```

Comment:
So the idea is to have various hooks in our Component's module parser. So far these are the ones I've thought might be needed. 
- after: happens after the whole parser has run, essentially if you just need to tack on some extra logic that isn't worried about intercepting or modify the DDO or ddo.controller.
- extendDDO: strictly for modifying the DDO before the call to angular.directive is made
- beforeCtrlInvoke: happens in the context of our controller 'hack' right before the real controller is invoked. Last chance to modify locals or other things.
- afterCtrlInvoke: Same as before but happens after Invoke.